File: tests/js/web-components.test.js
const fs = require("fs").promises;
const path = require("path");
const { expect, test } = require("@jest/globals");

test("top-nav-bar.js should have identical content", async () => {
  const furoFp = path.resolve(
    __dirname,
    "../../qiskit_sphinx_theme/furo/base/static/js/web-components/top-nav-bar.js"
  );
  const pytorchFp = path.resolve(
    __dirname,
    "../../qiskit_sphinx_theme/pytorch_base/static/js/web-components/top-nav-bar.js"
  );

  const furo = await fs.readFile(furoFp, "utf8");
  const pytorch = await fs.readFile(pytorchFp, "utf8");
  expect(furo).toEqual(pytorch);
});

File: tests/py/web_components_test.py
# This code is a Qiskit project.
#
# (C) Copyright IBM 2023.
#
# This code is licensed under the Apache License, Version 2.0. You may
# obtain a copy of this license in the LICENSE.txt file in the root directory
# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
#
# Any modifications or derivative works of this code must retain this
# copyright notice, and modified files need to carry a notice indicating
# that they have been altered from the originals.

"""Check that our vendored web components are identical between the different themes."""

from pathlib import Path


def test_top_nav_bar():
    furo = Path(
        "src/qiskit_sphinx_theme/theme/qiskit-sphinx-theme/static/js/web-components/top-nav-bar.js"
    ).read_text("utf-8")
    pytorch = Path(
        "src/qiskit_sphinx_theme/pytorch/static/js/web-components/top-nav-bar.js"
    ).read_text("utf-8")
    assert furo == pytorch
